tgui/packages/tgui/interfaces/Biogenerator.js

```diff
@@ -0,0 +1,247 @@
+import { useBackend, useSharedState } from "../backend";
+import {
+  Button,
+  Section,
+  Box,
+  Flex,
+  Icon,
+  Collapsible,
+  NumberInput,
+  ProgressBar,
+  Dimmer} from "../components";
+import { Window } from "../layouts";
+
+export const Biogenerator = (props, context) => {
+  const { data } = useBackend(context);
+  const { container } = data;
+
+  return (
+    <Window>
+      <Window.Content display="flex" className="Layout__content--flexColumn">
+        {!container ? (
+          <MissingContainer />
```

Comment:
So, this is a departure for the legacy UI behavior, but...
Given that the container is *only* necessary to create liquid outputs, and *not*  - what i have been assuming all these years until reviwing this PR - to hold the biomass... Can we have the "no container" splash replace only the Products list, and not the whole window?

Ideally, you'd be able to operate just fine without container at all, as long as you're not trying to make liquids (UI-wise: those exist, but are greyed out), but that's probably a bit too much work to support..

Reply:
Yeah I'll have it cover the product list instead.